### PR TITLE
C#: Update .NET 8 models.

### PR DESCRIPTION
**Pull request by michaelnebel**
### TLDR
In this PR we re-generate the .NET 8 runtime models based on the mixed model generation.

- A minor update for the `cs/information-exposure-through-exception` query is needed.
- To retain some query results we need to introduce manual models for `StringWriter`.
- When we originally added the generated models for .NET Runtime we didn't see that big changes to the produced alerts (maybe this is due to our DCA suite), so it was expected that we only would see few changes to the produced alerts, when updating the .NET 8 models. In any case, that we are able to retain all alerts with only some minor manual modelling and at the same time increase the precision of the models should be considered a success.

### Details

- It turns out that the `cs/information-exposure-through-exception` query relies on the imprecise `df-generated` summaries for taint propagation from an exception to its properties. With more precise summaries this is no longer possible. Instead we encode this directly in the query sources instead.
- The manual models for `StringWriter` is needed because they need to be used in conjunction with summaries for `TextWriter` (note that `StringWriter` extends `TextWriter`). More specifically the query results relied on the models seen below, but the content based models for `StringWriter` refers to the (synthetic) field containing a `StringBuilder` (as this is where the information is stored)
  - ["System.Text.Encodings.Web", "TextEncoder", False, "Encode", "(System.IO.TextWriter,System.String)", "", "Argument[1]", "Argument[0]", "taint", "df-generated"]
  - ["System.IO", "StringWriter", False, "ToString", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]

DCA results

- Peformance looks good.
- The new results for `cs/exposure-of-sensitive-information`, `cs/information-exposure-through-exception` and `cs/sensitive-data-transmission` on `mono` are due to the manual models for `System.IO.StringWriter`.

### CODE REVIEW

File: csharp/ql/test/library-tests/dataflow/collections/CollectionFlow.expected [hunk 1]
```diff
@@ -24,6 +24,7 @@ models
 | 23 | Summary: System; Span<T>; false; Span; (T); ; Argument[0]; Argument[this].Element; value; manual |
 | 24 | Summary: System; Span<T>; false; Span; (T[]); ; Argument[0].Element; Argument[this].Element; value; manual |
 | 25 | Summary: System; Span<T>; false; ToArray; (); ; Argument[this].Element; ReturnValue.Element; value; manual |
+| 26 | Summary: System.Collections.Generic; List<T>+Enumerator; false; get_Current; (); ; Argument[this].Property[System.Collections.Generic.List`1+Enumerator.Current]; ReturnValue; value; dfc-generated |
```
Comment (hvitved) [inline csharp/ql/test/library-tests/dataflow/collections/CollectionFlow.expected:27]: This looks like a summary we shouldn't generate: It is always the case that `get_Foo` induces a readstep from `Argument[this]` to `ReturnValue`.

Comment (michaelnebel) [reply csharp/ql/test/library-tests/dataflow/collections/CollectionFlow.expected:27]: I think that when we discussed it last (when you made https://github.com/github/codeql/pull/16088) - we agreed to keep model generation for properties that wasn't read/write (in this case the property only has a getter).
Do you think we should revisit this decision?

File: csharp/ql/lib/change-notes/2024-10-07-netruntime-models.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+category: majorAnalysis
+---
+* C#: The generated .NET 8 runtime models have been updated.
```
[the remark this answers is not in the document]
Comment (michaelnebel) [reply csharp/ql/lib/change-notes/2024-10-07-netruntime-models.md:4]: I will remove it.

File: csharp/ql/test/library-tests/dataflow/collections/CollectionFlow.expected (hunk 1 above)
Comment (hvitved) [reply csharp/ql/test/library-tests/dataflow/collections/CollectionFlow.expected:27]: I think that if we generate a summary for a getter that is saying exactly that it is a getter, then there is no point (since we already have a generic read step for that).

Comment (michaelnebel) [reply csharp/ql/test/library-tests/dataflow/collections/CollectionFlow.expected:27]: This is also not something that we generally do.
The example in question is because there are two property implementations for `Current`, where the concrete interface implementation returns the value of the other `Current` property. The other "known" example where this can happen is for recursive property definitions (which do exist - but I sure hope that this is not a general programming pattern).

Comment (hvitved) [reply csharp/ql/test/library-tests/dataflow/collections/CollectionFlow.expected:27]: Makes sense, thanks a lot for the explanation.

**Review by hvitved — APPROVED** [review]
